types/block.go

```diff
@@ -1169,7 +1176,10 @@ func DataFromProto(dp *tmproto.Data) (Data, error) {
 	if len(dp.Blobs) > 0 {
 		blobs := make([]Blob, len(dp.Blobs))
 		for i, m := range dp.Blobs {
-			blobs[i] = Blob{NamespaceID: m.NamespaceId, Data: m.Data}
+			if m.ShareVersion > math.MaxUint8 {
+				return Data{}, fmt.Errorf("share version %d is too large", m.ShareVersion)
+			}
+			blobs[i] = Blob{NamespaceID: m.NamespaceId, Data: m.Data, ShareVersion: uint8(m.ShareVersion)}
 		}
 		data.Blobs = blobs
 	} else {
```

Comment:
nit: since we are touching this, we could simplify and reduce some locs by removing the if statement entirely.
```go
blobs := make([]Blob, len(dp.Blobs))
for i, m := range dp.Blobs {
   ...
}
data.Blobs = blobs
```

Reply:
Good idea! Resolved in d9edeaae7d8687a7d560e699e2c56acc4f559fa7